ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,852 @@
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.localEnvironment.artifactDAO.addSiteLocation(artifact, new SiteLocation(randomSiteID));
+
+        try {
+            System.setProperty("user.home", TMP_DIR);
+            InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI,
+                                                                    new IncludeArtifacts(),null,
+                                                                    true);
+            testSubject.run();
+        } finally {
+            System.setProperty("user.home", USER_HOME);
+        }
+
+        Artifact localArtifact = this.localEnvironment.artifactDAO.get(artifact.getID());
+        Assert.assertNotNull("local artifact not found", localArtifact);
+        Assert.assertEquals("metaChecksum mismatch", artifact.getMetaChecksum(), localArtifact.getMetaChecksum());
+        Assert.assertTrue("artifact does not contains remote site location",
+                          localArtifact.siteLocations.contains(new SiteLocation(remoteSiteID)));
+        Assert.assertTrue("artifact does not contains local site location",
+                          localArtifact.siteLocations.contains(new SiteLocation(randomSiteID)));
+    }
+
+    /**
+     * explanation6: deleted from L, lost DeletedArtifactEvent
+     * evidence: ?
+     * action: assume explanation3
+     *
+     * explanation7: L==storage, deleted from L, lost DeletedStorageLocationEvent
+     * evidence: ?
+     * action: assume explanation3
+     *
+     * explanations 6 and 7 are covered by explanation 3
+     */
+
+    /**
+     * discrepancy: artifact.uri in both && artifact.id mismatch
+     *
+     * explanation1: same ID collision due to race condition that metadata-sync has to handle
+     * evidence: no more evidence needed
+     * action: pick winner, create DeletedArtifactEvent for loser, delete loser if it is in L
+     *
+     * case 1:
+     * local Artifact contentModifiedDate before remote Artifact contentModifiedDate, delete local Artifact.
+     * before: Artifact.uri in L && R, L Artifact.id != R Artifact.id,
+     *         L Artifact.contentModifiedDate < R Artifact.contentModifiedDate
+     * after:  L Artifact deleted, R Artifact in L, DeletedArtifactEvent in L
+     *
+     * case 2:
+     * local Artifact contentModifiedDate after remote Artifact contentModifiedDate, no action
+     * before: Artifact.uri in L && R, L Artifact.id != R Artifact.id,
+     *         L Artifact.contentModifiedDate >= R Artifact.contentModifiedDate
+     * after:  Artifact in L
+     */
+    @Test
+    public void artifactUriCollision_LocalIsStorage() throws Exception {
+        artifactUriCollision(false);
+    }
+
+    @Test
+    public void artifactUriCollision_LocalIsGlobal() throws Exception {
+        artifactUriCollision(true);
+    }
+
+    public void artifactUriCollision(boolean trackSiteLocations) throws Exception {
```

Comment:
The setup for this test is still more complicated than it needs to be and hence to harder than necessary to verify correctness.

First, if you just do `new Artifact(URI, URI, Date)` it always assigns a new UUID. basically, `new Artifact` does exactly what it says: creates a new entity. Manually creating and assigning UUID is misleading because it looks like trying to finesse something different than that. (In CAOM, I never had that ctor at all and assigned entity.id via reflection... this was meant as a convenience for deserialisation code that other applications should never use)

Second, it would be more clear if you create the two test artifacts in the order implied by the scenario. So in case 1 create the localArtifact first then the remote.

Third, Calendar.roll(...) requires a javadoc lookup to see what true|false means. If you did something like this:
```
Date now = new Date();
Date older = new Date(now.getTime() - 2000L);
```
and then used those two values in the ctors it would be much more clear. I thought about now and "newer" but the simplest way to do that gives a contentLastModified in the future and that seems like a strange thing that future validation constraints might reject. I like test scenarios to be normal/realistic.

Given the complexity/subtlety of the logic in this test class it's important to not have distracting